OCPBUGS-38489: ztp: remove unsupported KlusterletAddonConfig clusterLabels and update managecluster CR to include cloud and vendor label (#2005)
openshift-kni/cnf-features-deploy: ztp/gitops-subscriptions/argocd/README.md

    $ oc patch multiclusterengines.multicluster.openshift.io multiclusterengine --type=merge --patch-file out/argocd/deployment/disable-cluster-proxy-addon.json
```

> 5. Optionally: If configuring an existing hub cluster, starting ACM 2.10 apply the following patch to allow ACM to update `vendor` and `cloud` labels in ManagedCluster CR if needed.

Reviewer: ```suggestion
5. Optionally: If configuring an existing hub cluster, starting with ACM 2.10, the following patch is needed to allow ACM to update `vendor` and `cloud` labels in ManagedCluster CRs. This enables observability.
```
or something along those lines.

Author: Updated

    oc apply -k out/argocd/deployment
```

7. Optionally: If configuring an existing hub cluster (i.e skipped step 6, pipeline configuration), starting with ACM 2.10, the following patch is needed to allow ACM to update `vendor` and `cloud` labels in ManagedCluster CRs. This enables observability without user intervention.
```

>     $ oc patch applications.argoproj.io clusters -n openshift-gitops --type='json' --patch-file out/argocd/deployment/allow-acm-to-override-vendor-and-cloud-value.json

Reviewer 2: Does it mean that patching this json file will add those labels under the ManagedCLusterCR for the existing deployment?

Author: Yeah essentially! So the labels will be there in the ManagedCluster CR right away. But with this patch...argocd will allow "vendor: auto-detect" to update to appropriate value (e.g "vendor: OpenShift"). 

Without this patch argocd will not sync since by default it always tries to match all the content that's in git with the live cluster. So ACM tries to update and Argocd reverts
